File: 1894-merge-strings-alternately/1894-merge-strings-alternately.go
package main

import "fmt"

func mergeAlternately(word1 string, word2 string) string {
	ans := ""
	word1Idx := 0
	word2Idx := 0

	for i := 0; i < min(len(word1), len(word2)); i++ {
		ans += string(word1[word1Idx])
		word1Idx++
		ans += string(word2[word2Idx])
		word2Idx++
	}

	for i := word1Idx; i < len(word1); i++ {
		ans += string(word1[i])
	}

	for i := word2Idx; i < len(word2); i++ {
		ans += string(word2[i])
	}
	return ans
}

File: 1894-merge-strings-alternately/1894-merge-strings-alternately.js
/**
 * @param {string} word1
 * @param {string} word2
 * @return {string}
 */
var mergeAlternately = function (word1, word2) {
    let ans ="";
    let lenWord1 = word1.length, lenWord2 = word2.length;
    let pointerWord1 = 0, pointerWord2 = 0;
    while (lenWord1 > 0 && lenWord2 > 0) {
        ans +=(word1[pointerWord1] + word2[pointerWord2]);
        lenWord1--
        lenWord2--
        pointerWord1++
        pointerWord2++
    }

    while (lenWord1>0) {
        lenWord1--;
        ans += (word1[pointerWord1++]);
    }

    while (lenWord2>0) {
        lenWord2--;
        ans += (word2[pointerWord2++]);
    }
    return ans;
};